probe-rs/src/architecture/arm/sequences/mod.rs

```diff
@@ -91,9 +91,22 @@ pub trait ArmDebugSequence: Send + Sync {
         // Ensure current debug interface is in reset state.
         interface.swj_sequence(51, 0x0007_FFFF_FFFF_FFFF)?;
 
-        // Execute SWJ-DP Switch Sequence JTAG to SWD (0xE79E).
-        // Change if SWJ-DP uses deprecated switch code (0xEDB6).
-        interface.swj_sequence(16, 0xE79E)?;
+        match interface.active_protocol() {
+            Some(crate::WireProtocol::Jtag) => {
+                // Execute SWJ-DP Switch Sequence SWD to JTAG (0xE73C).
+                interface.swj_sequence(16, 0xE73C)?;
+            }
+            Some(crate::WireProtocol::Swd) => {
+                // Execute SWJ-DP Switch Sequence JTAG to SWD (0xE79E).
+                // Change if SWJ-DP uses deprecated switch code (0xEDB6).
+                interface.swj_sequence(16, 0xE79E)?;
+            }
+            _ => {
+                return Err(crate::Error::Probe(DebugProbeError::NotImplemented(
```

Comment:
I am not sure if it's correct to fail here. Can't the `SWJ-DP Switch Sequence` always be executed for example, I thought so. If the probe would be stuck in unknown state for whatever reason and we cannot read the current protocol, it would be stuck there forever.

Reply:
The idea is that if we don't have a protocol selected there is no way to generate a correct SWJ sequence as we don't know what we're switching to.  The only alternative I could think to do is to default to either SWD or JTAG if not specified.

We could panic as getting here without a protocol error selected will never work / recover and is a usage error.

Reply:
> We could panic as getting here without a protocol error selected will never work / recover and is a usage error.

@thefaxman I would prefer it if we never panic. VSCode users, via the probe-rs-debugger, won't see panics. I would prefer if we can propogate a useful error that the UX can catch and help the user be more successful next time :)

Reply:
> I would prefer it if we never panic.

I think that makes sense, the question is then should the current logic that returns an error remain or not.  The only other option I could see to do is to default to a protocol, but I think that's the choice of the probe and not of this layer as if they disagree this will just break.

Reply:
Hmm that's fair. The name is missleading then I think. `active_protocol` - to me - suggests that it's the currently active protocol, you are switching **from** not **to**.
So the code makes sense to me, maybe the name not so much in this context. In general it makes sense tho.
So let's leave as is.

Reply:
I can add a comment here indicating that the goal is to ensure the DAP is speaking the protocol selected via active_protocol, regardless of any current DAP state.  Would that help clarify the goal?

Reply:
Yeah that would have been good, but as always I was a bit trigger happy. Can do it retroactively if you feel like :)

Reply:
No problem - I'll put the comment in my next batch of changes.  I'd rather commit early and often.